packages/graphql-language-service-interface/src/getAutocompleteSuggestions.ts

```diff
@@ -815,6 +890,13 @@ export function getTypeInfo(
         if (state.name) {
           type = schema.getType(state.name);
         }
+        // TODO: collect already extended interfaces of the type/interface we're extending
+        //  here to eliminate them from the completion list
+        // because "type A extends B & C &" should not show completion options for B & C still.
+
+        // if(state.prevState?.kind === RuleKinds.IMPLEMENTS) {
+        //   interfaceDef?.addInterface(new GraphQLInterfaceType({ name: state.name}))
```

Comment:
@IvanGoncharov this is the area I was talking about, not sure how best to do this other than cloning the interfaceDef instead of being able to mutate it?

Reply:
@danielrearden this is where I need to clone and re-create the interface type, but I also need to determine when the interface definition is complete. I think this is when state.kind === `INTERFACE_DEF` AND step === 2? so then i can do interfaceDef = null